### PR TITLE
Yahya/features/base terraform

### PR DESCRIPTION
**Pull request by yahya130**
- Pipeline still to be tested (fails on terraform init for a weird reason at the moment, will create a separate issue for this bug for the next iteration (3) 
- Bear in mind that action should fail at the moment because azure is not fully setup yet

### CODE REVIEW

**Comment by martinkearn** [comment]
Would like to see this demonstrated at a stand-up parking lot if possible?

**Review by martinkearn — APPROVED** [review]

File: terraform/environments/backend/dev.conf [hunk 1]
```diff
@@ -0,0 +1,5 @@
+resource_group_name  = "TBC"
```
[the remark this answers is not in the document]
Comment (yahya130) [reply terraform/environments/backend/dev.conf]: Updated the naming, thanks

**Comment by yahya130** [comment]
> Would like to see this demonstrated at a stand-up parking lot if possible?

That is a good idea, I can do, pipeline is failing at the moment due to minor issues, let me get this working first

File: .github/workflows/terraform-deploy-template.yml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+      tf_vars: ${{ inputs.tf_vars }}
+      backend_config: ${{ inputs.backend_config }}
+
+    steps:
+    # Checkout the repository to the GitHub Actions runner
+    - name: Checkout
+      uses: actions/checkout@v3
+    
+    - name: Setup Terraform
+      uses: hashicorp/setup-terraform@v1
```
Comment (guypritchard) [inline .github/workflows/terraform-deploy-template.yml:52]: There's a version 2 of this now.  I don't know what that gets you - but might be worth trying that out?

File: .github/workflows/terraform-deploy-template.yml [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+ 
+    # Checks that all Terraform configuration files adhere to a canonical format
+    - name: Terraform Format
+      run: terraform fmt -check
+         
+    # Initialize a new or existing Terraform working directory by creating initial files, loading any remote state, downloading modules, etc.
+    - name: Terraform Init
+      run: terraform init -backend-config=$backend_config
+    
+    # Ensures all the terraform code and files are syntaxically correct
```
Comment (guypritchard) [inline .github/workflows/terraform-deploy-template.yml:62]:
```suggestion
    # Ensures all the terraform code and files are syntactically correct 
```

**Review by guypritchard — COMMENTED** [review]
It would be great to see a demo of this so we can understand how this fits together with environmental configuration etc...

**Pull request merged**